From d2122a0d2278e5c621a601a3ec90d4790a784715 Mon Sep 17 00:00:00 2001
From: Neal Gompa <ngompa13@gmail.com>
Date: Thu, 24 Nov 2016 08:02:48 -0500
Subject: [PATCH] Add path for libstemmer headers to be found

In Mageia, the libstemmer headers are in /usr/include/libstemmer,
while in Fedora and other distributions, they are in /usr/include.

We will now support both paths.
---
 src/CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 1132d6c..3d8b2f6 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -8,12 +8,14 @@ find_library(M_LIB m)
 pkg_check_modules(YAML REQUIRED yaml-0.1)
 
 if (STEMMING)
+	find_path(LIBSTEMMER_INCLUDE_DIR libstemmer.h HINTS "/usr/include" "/usr/include/libstemmer")
 	find_library(SNOWBALL_LIB stemmer)
-	if (NOT SNOWBALL_LIB)
+	if (NOT SNOWBALL_LIB OR NOT LIBSTEMMER_INCLUDE_DIR)
 		message (FATAL_ERROR "Stemming support is enabled, but libstemmer was not found. Please install the Snowball development files.")
 	endif ()
 else ()
 	set(SNOWBALL_LIB "")
+	set(LIBSTEMMER_INCLUDE_DIR "")
 endif ()
 
 include(GObjectIntrospectionMacros)
@@ -99,12 +101,14 @@ set(APPSTREAM_LIB_PRIVATE_HEADERS
 		as-stemmer.h
 )
 
+
 include_directories(${CMAKE_BINARY_DIR}
 		    ${CMAKE_BINARY_DIR}/src
 		    ${GLIB2_INCLUDE_DIRS}
 		    ${GIO2_INCLUDE_DIRS}
 		    ${LIBXML2_INCLUDE_DIR}
 		    ${YAML_INCLUDE_DIR}
+		    ${LIBSTEMMER_INCLUDE_DIR}
 )
 
 add_definitions("-DAS_COMPILATION")
-- 
2.9.2

